snapshotter/Makefile

```diff
@@ -11,12 +11,14 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-all: naive
+SUBDIRS:=cmd/devmapper cmd/naive
 
-naive: *.go
-	go build -o naive_snapshotter ./cmd/naive
+all: $(SUBDIRS)
+
+$(SUBDIRS):
+	$(MAKE) -C $@
 
 clean:
-	- rm -f naive_snapshotter
+	for d in $(SUBDIRS); do $(MAKE) -C $$d clean; done
 
-.PHONY: all clean
+.PHONY: all $(SUBDIRS) clean proto
```

Comment:
Didn't pay attention to this. What's the difference?

Reply:
I think mostly convention because some things expect newlines at the end of every line.
See first answer [here](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) for more detail.

Reply:
Does it break `make` command?

Reply:
No....

Reply:
Missed this one. Sounds like not very important, will address in subsequent PRs.

Reply:
There are a couple areas where this is annoying. Some editors, including vim, will automatically insert missing trailing newlines. This can result in unexpected modifications to files (granted, very minor).

It's also annoying when using `cat` to view a file; a missing trailing newline means that the next command prompt is presented on the same line as the last line of the file. This is often confusing and always ugly.